File: backend/ManageMe/Database/Models/Task.cs
using ManageMe.Enums;

namespace ManageMe.Database.Models;

public class Task
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public Priority Priority { get; set; }
    public State State { get; set; }
    public int ExpectedTime { get; set; }
    public DateTime CreatedOn { get; set; }
    public DateTime? StartedOn { get; set; }
    public DateTime? EndedOn { get; set; }
    public string? AssignedUser { get; set; }
    public int UniqueEntityId { get; set; }
}

File: manage-me/src/app/models/Task.ts
import { Priority, State } from "./Feature";

export class Task {
    id?: number;
    name: string;
    description: string;
    priority: Priority;
    state: State;
    expectedTime: number;
    createdOn: Date;
    startedOn?: Date;
    endedOn?: Date;
    assignedUser?: string;
    featureId: number;

    constructor(name: string, description: string, priority: Priority, state: State, expectedTime: number, createdOn: Date, featureId: number,
        id?: number, startedOn?: Date, endedOn?: Date, assignedUser?: string) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.priority = priority;
        this.state = state;
        this.expectedTime = expectedTime;
        this.createdOn = createdOn;
        this.featureId = featureId;
        this.startedOn = startedOn;
        this.endedOn = endedOn;
        this.assignedUser = assignedUser;
    }
}